sevntu-checks/src/main/java/com/github/sevntu/checkstyle/checks/design/ConstructorWithoutParamsCheck.java

```diff
@@ -0,0 +1,158 @@
+////////////////////////////////////////////////////////////////////////////////
+
+package com.github.sevntu.checkstyle.checks.design;
+
+import java.util.regex.Pattern;
+
+import com.puppycrawl.tools.checkstyle.api.Check;
+import com.puppycrawl.tools.checkstyle.api.DetailAST;
+import com.puppycrawl.tools.checkstyle.api.TokenTypes;
+import com.puppycrawl.tools.checkstyle.utils.CommonUtils;
+
+/**
+ * <p>
+ *   This check prohibits usage of parameterless constructors,
+ *   including the default ones.
+ * </p>
+ *
+ * <p>
+ * <b>Rationale:</b> constructors of certain classes must always take arguments
+ * to properly instantiate objects. Exception classes are the primary example:
+ * their objects must contain enough info to find out why an exception occurred
+ * (see "Effective Java", item 63). Constructing an exception without a cause exception
+ * or an exception message leaves out such info and thus should be prohibited.
+ * </p>
+ *
+ * <p>
+ *     <b>Default configuration:</b>
+ *     <pre>
+ *     {@code
+ *         < module name="ConstructorWithoutParamsCheck" >
+ *           <property name="classNameFormat" value=".*Exception"/>
+ *         < /module >
+ *     }
+ *     </pre>
+ * </p>
+ *
+ * <p>
+ * <b>Examples:</b>
+ *
+ * <pre>
+ * // the check can prohibit default constructors of built-in classes
+ * RuntimeException ex = new RuntimeException(); // prohibited
+ *
+ * // the check allows constructors with empty arguments
+ * RuntimeException ex = new RuntimeException(""); // allowed
+ *
+ * // the check can prohibit default constructors of user-defined classes
+ * public class Clazz1 {
+ * }
+ *
+ * Clazz1 o1 = new Clazz1(); // prohibited
+ *
+ * // the check can prohibit  user-defined parameterless constructors
+ * public class Clazz2 {
+ *
+ *   Clazz2() {
+ *     foobar();
+ *   }
+ *
+ * }
+ *
+ * Clazz2 o2 = new Clazz2(); // prohibited
+ *
+ * </pre>
+ *
+ * </p>
+ * For more examples , see InputConstructorWithoutParamsCheck.
+ * For discussion, see the sevntu-checkstyle
+ * <a href=https://github.com/sevntu-checkstyle/sevntu.checkstyle/issues/412> issue 412</a>.
+ *
+ * @author <a href="mailto:Sergey Dudoladov@gmail.com">Sergey
+ *         Dudoladov</a>
+ */
+public class ConstructorWithoutParamsCheck extends Check {
+
+    /**
+     * This key points to the warning message in the  "messages.properties" file.
+     */
+    public static final String MSG_KEY = "constructor.without.params";
+
+    /**
+     * The format string of the regexp.
+     */
+    private String classNameFormat = ".*Exception";
+
+    /**
+     * The regexp to match against.
+     */
+    private Pattern regexp = Pattern.compile(classNameFormat);
+
+    @Override
+    public int[] getDefaultTokens() {
+        return new int[] {TokenTypes.LITERAL_NEW};
+    }
+
+    @Override
+    public int[] getRequiredTokens() {
+        return getDefaultTokens();
+    }
+
+    @Override
+    public int[] getAcceptableTokens() {
+        return getDefaultTokens();
```

Comment:
@romani Cobertura reports this method call (line 120) to be the only one not covered by tests. Which code is supposed to call it ?  The user of this check is not supposed to set any other tokens besides LITERAL_NEW.